core/lib/spree/testing_support/factory_bot.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require "factory_bot"
+begin
+  require "factory_bot_rails"
+rescue LoadError
+end
+
+module Spree
+  module TestingSupport
+    module FactoryBot
+      SEQUENCES = ["#{::Spree::Core::Engine.root}/lib/spree/testing_support/sequences.rb"]
+      FACTORIES = Dir["#{::Spree::Core::Engine.root}/lib/spree/testing_support/factories/**/*_factory.rb"].sort
+      PATHS = SEQUENCES + FACTORIES
+
+      def self.definition_file_paths
+        @paths ||= PATHS.map { |path| path.sub(/.rb\z/, '') }
+      end
+
+      def self.deprecate_cherry_picking
+        # All good if the factory is being loaded by FactoryBot.
+        return if caller.find { |line| line.include? "/factory_bot/find_definitions.rb" }
+
+        Spree::Deprecation.warn(
+          "Please do not cherry-pick factories, this is not well supported by FactoryBot. " \
+          'Use `require "spree/testing_support/factories"` instead.', caller(2)
```

Comment:
`caller(1)` is giving me more accurate details when running specs in a real store: 

```
# caller(2) example result
DEPRECATION WARNING: Please do not cherry-pick factories, this is not well supported by FactoryBot. Use `require "spree/testing_support/factories"` instead. (called from block in <top (required)> at /Users/kennyadsl/Code/nebulab/solidus/core/lib/spree/testing_support/factories.rb:11)

# caller(1) example result
DEPRECATION WARNING: Please do not cherry-pick factories, this is not well supported by FactoryBot. Use `require "spree/testing_support/factories"` instead. (called from <top (required)> at /Users/kennyadsl/Code/nebulab/solidus/core/lib/spree/testing_support/factories/zone_factory.rb:4)

```

Reply:
I updated the code to mention the changelog instruction as a way to fix the deprecation and to point it to the first non-core line from the backtrace so ideally it will point right in the place where the cherry-picking is happening (i.e. inside the rails application).